src/DataflowPass.cpp

```diff
@@ -108,6 +147,7 @@ void DataflowPass::transfer(Instruction *instruction,
     for (unsigned i = 0; i < Call->getNumArgOperands(); ++i) {
       ProgramVariable argumentVar = ProgramVariable(Call->getArgOperand(i));
       std::string arg = argumentVar.getCleanedName();
+      ProgramVariable *pv = inputProgramPoint.getPVRef(arg, false);
```

Comment:
This isn't used until over 20 lines later, at line 172. Is there a good reason to compute it so early? I don't see one immediately.

Reply:
The same `pv` ref is used in many other spots in the scope it's defined in, so I'd rather initialize `pv` then refer to it when we need it, rather than repeatedly initialize it in all the inner scopes. The difference would pretty much be doing this: (pseudocode)  

```
pv = input(M) 

if (a) { 
  use1(pv)
  continue
}

if (b) { 
  use2(pv)
  continue
}

```

instead of this: 
```
if (a) { 
  pv = input(M) 
  use(pv)
  continue
}

if (b) { 
  pv = input(M)
  use(pv)
  continue
}

```
and I prefer the former design. But, let me know if I actually misinterpreted your comment, or if the latter design is preferable.

Reply:
I see what you mean. Fundamentally, the issue here is really just that this method is too big: it doesn't all fit onto a screen, and so to read it I have to constantly scroll up and down. This means that it's hard to see where variables come from when they are used.

For methods like this, I usually try to write helper methods for the parts inside the various `if` statements. So, the psuedocode would become:
```
pv = input(M) 

if (a) { 
  aLogic(pv, ...);
}

if (b) { 
  bLogic(pv, ...);
}
```

This lets a reader see the high-level structure easily, and then the reader can dive into each part on its own.